libs/vre/shared/app-resource-properties/src/lib/upload-file.service.ts

```diff
@@ -1,46 +1,60 @@
-import { HttpClient, HttpParams } from '@angular/common/http';
+import { HttpClient, HttpHeaders } from '@angular/common/http';
 import { Injectable } from '@angular/core';
 import { AppConfigService } from '@dasch-swiss/vre/shared/app-config';
 import { AccessTokenService } from '@dasch-swiss/vre/shared/app-session';
 import { Observable } from 'rxjs';
+import { map } from 'rxjs/operators';
 
-export interface UploadedFile {
-  fileType: string;
-  internalFilename: string;
-  originalFilename: string;
-  temporaryUrl: string;
+interface UploadedFileResponse {
+    internalFilename: string;
 }
 
-export interface UploadedFileResponse {
-  uploadedFiles: UploadedFile[];
+export interface UploadedFile {
+    internalFilename: string;
+    thumbnailUrl: string;
+    baseUrl: string;
 }
 
 @Injectable({
-  providedIn: 'root',
+    providedIn: 'root',
 })
 export class UploadFileService {
-  constructor(
-    private readonly _acs: AppConfigService,
-    private readonly _http: HttpClient,
-    private readonly _accessTokenService: AccessTokenService
-  ) {}
+    constructor(
+        private readonly _acs: AppConfigService,
+        private readonly _http: HttpClient,
+        private readonly _accessTokenService: AccessTokenService,
+    ) {
+    }
 
-  /**
-   * uploads files to SIPI
-   * @param (file)
-   */
-  upload(file: FormData): Observable<UploadedFileResponse> {
-    const uploadUrl = `${this._acs.dspIiifConfig.iiifUrl}/upload`;
+    /**
+     * uploads files to ingest server
+     * @param (file) The file to upload
+     * @param (shortcode) The project shortcode
+     */
+    upload(file: File, shortcode: string): Observable<UploadedFile> {
 
-    const jwt = this._accessTokenService.getAccessToken()!;
-    const params = new HttpParams().set('token', jwt);
+        const jwt = this._accessTokenService.getAccessToken()!;
+        const headers = new HttpHeaders({
+                'Content-Type': 'application/octet-stream}',
+                'Authorization': `Bearer ${jwt}`,
+            },
+        );
 
-    // --> TODO in order to track the progress change below to true and 'events'
-    const options = {
-      params,
-      reportProgress: false,
-      observe: 'body' as const,
-    };
-    return this._http.post<UploadedFileResponse>(uploadUrl, file, options);
-  }
+        const options = {
+            reportProgress: false,
+            observe: 'body' as const,
+            headers: headers,
+        };
+        const url = `${this._acs.dspIngestConfig.url}/projects/${shortcode}/assets/ingest/${file.name}`;
+        return this._http.post<UploadedFileResponse>(url, file, options).pipe(
+            map((res: UploadedFileResponse) => {
+                let baseUrl = `${this._acs.dspIiifConfig.iiifUrl}/${shortcode}/${res.internalFilename}`;
+                return {
+                    internalFilename: res.internalFilename,
+                    thumbnailUrl: `${baseUrl}/full/256,/0/default.jpg`,
```

Comment:
make sure it is correct, throws exceptions locally:
       GET https://iiif.dev.dasch.swiss/0811/0FJ4CJ0QFDD-tTYHo455xhK.jpx/full/256,/0/default.jpg 404 (Not Found)

Reply:
This PR is not complete yet and also you will have to run Sipi with the changes from https://github.com/dasch-swiss/dsp-api/pull/3250 which is not integrated yet.